src/Components/CreateImageWizardV2/steps/Details/index.tsx

```diff
@@ -27,13 +28,28 @@ const DetailsStep = () => {
   const dispatch = useAppDispatch();
   const blueprintName = useAppSelector(selectBlueprintName);
   const blueprintDescription = useAppSelector(selectBlueprintDescription);
+
   const handleNameChange = (
     _event: React.FormEvent<HTMLInputElement>,
     name: string
   ) => {
     dispatch(changeBlueprintName(name));
   };
 
+  const { data: sameNameBlueprint, isFetching } = useGetBlueprintsQuery(
+    { name: blueprintName },
+    { skip: blueprintName.length < 2 }
+  );
```

Comment:
We're firing for every name change, which is bit unfortunate, but I didn't manage to make debounce work for this.

Reply:
have you tried something like:

```js
  const [debouncedBlueprintName, setDebouncedBlueprintName] = useState(blueprintName);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedBlueprintName(blueprintName);
    }, 500); 
    return () => {
      clearTimeout(handler);
    };
  }, [blueprintName]);

  const { data: sameNameBlueprint, isFetching } = useGetBlueprintsQuery(
    { name: debouncedBlueprintName },
    { skip: debouncedBlueprintName.length < 2 }
  );
```

Reply:
I haven't, I guess this would work, but it would be setting the name to the local state on top of the global one.
Tho I guess it's simpliest solution we have now.

Reply:
I've fixed this in new version, but unfortunatelly it requires the caller to provide a memoized function, so it's prone to mistakes.